### ISSUE TITLE
Need to expose Kubelet --config flag in KubeletConfigSpec

### ISSUE THREAD

**Issue by tvi** [kubernetes/kops#5070]
We are running 1.10.1 cluster and our logs contain following log line
```
kubelet[8594]: Flag --max-pods has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
```
The problem is that we cannot specify --config file and KubeletConfiguration spec.

**Comment by ihoegen** [comment]
I'm not seeing it exposed in 

https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/componentconfig.go

so it'll have to be added.

It's super easy to expose, see:

https://github.com/kubernetes/kops/blob/master/docs/development/api_updates.md

Add the struct value, generate api machinery and you'll be good to

**Comment by sdanbury** [comment]
/remove-lifecycle rotten

I am currently hitting this. This is what I see when I `ps aux | grep kubelet` on one of the worker nodes, so I can confirm kops is setting the flag correctly:
```
/opt/kubernetes/bin/kubelet --feature-gates=ExperimentalCriticalPodAnnotation=true...
```

But then in the logs you see the same messages shown above, one of those being:
```
Flag --feature-gates has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/
```

Did anybody get anywhere on this or should I take a look?

**Comment by evanmcclure** [comment]
/remove-lifecycle rotten

Any updates?

**Comment by mars64** [comment]
/remove-lifecycle rotten

I think this is still a desirable feature.

**Comment by rifelpet** [comment]
Agreed, given that kubelet is deprecating the old style of providing its configuration I definitely think we should migrate to defining the config file automatically.

/lifecycle frozen
